### DISCUSSION TITLE
API for Extension Authors

### DISCUSSION THREAD

**Discussion by AdamRaichu** [lunapaint/vscode-luna-paint#212] [Ideas]
Hey, I love your extension, and use it regularly.

I would love it if your extension could return an API. I am the author of [this extension](https://github.com/AdamRaichu/vscode-zip-viewer) which is for previewing the contents of zip files. It would be great if I could use an API from your extension to open your editor with the image from my zip file. Most importantly though, the image would not exist on the filesystem unless the user saved.

In short, my request is that you would return a function from your `exports.activate` that takes a Uint8Array (the image contents) and opens your editor with it.

I would be happy to help contribute to this, (and already would've started if it wasn't closed source).

**Comment by Tyriar** [comment]
I think you can achieve this already by opening a data URI of the image as that should be routed to open using the correct custom editor. Have you tried that and it didn't work?

**Reply by AdamRaichu** [reply]
Didn't work with this test code. Did I misunderstand?

```js
const vscode = require("vscode");
    const imageUri = vscode.Uri.parse(
      "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABUAAAAYCAMAAAAiV0Z6AAAAPFBMVEVLoEN0wU6CzFKCzFKCzFKCzFKCzFJSo0MSczNDmkCCzFJPoUMTczNdr0gmgziCzFITczMTczMTczMTczPh00jOAAAAFHRSTlPF/+bIsms8Ad///hX+//5/tXw7aMEAx10AAACaSURBVHgBbc4HDoRQCATQ33tbvf9dF9QxaCT9UQaltLHOh/golXKhMs5Xqa0xU1lyoa2fXFyQOsDG38qsLy4TaV+sFislovyhPzLJJrBu6eQOtpW0LjbJkzTuTDLRVNKa3uxJI+VdiRqXSeu6GW+Qxi29eLIi8H7EsYrT42BD+mQtNO5JMjRuC4lSY8V4hsLX0egGijvUSEP9AbylEsOkeCgWAAAAAElFTkSuQmCC"
    );
    vscode.commands.executeCommand("vscode.open", imageUri);
```

![image](https://github.com/lunapaint/vscode-luna-paint/assets/91494019/c7419317-ccb4-478e-aa58-17e09988eabd)

**Reply by Tyriar** [reply]
Hmm apparently this isn't supported for custom editors ☹️. It was prototyped to include a mime type in the custom editor selector but ended up not happening.

I'd recommend persisting the file to a temp folder on disk then. If your extension allows saving you could listen to the temp file changing and recompress the archive on change?

**Reply by AdamRaichu** [reply]
For a normal extension I think that would work great! However, I am trying to keep this extension compatible with the web. Ultimately, I think I'll stick with the system I have currently, which is to just display the image in a webview. Thank you for your time though, and happy coding!
